### ISSUE TITLE
AES - OCB does not work for 15 bytes nonces (default length)

### ISSUE THREAD

**Issue by user163** [Legrandin/pycryptodome#664]
The PyCryptodome implementation for AES in OCB mode gives wrong results for 15 bytes nonces (the default length). The 15th byte of the nonce is ignored and effectively the corresponding 14 bytes nonce is used for processing. 

All test vectors of [RFC 7253 The OCB Authenticated-Encryption Algorithm][1] have only a 12 bytes nonce, so 15 bytes nonces are not testable with this.<br>
On this [website][2] you can find more OCB test vectors, especially [*test-vector-1-nonce120.txt*][3]. 

To reproduce the problem here, the 4th test vector is used:

```none
K: 000102030405060708090A0B0C0D0E0F
N: EEDDCCBBAA99887766554433221103
A:
P: 0001020304050607
C: 084E869570194BD25032FE9E5328E45D507E74F3366E20D2

with 

K: key, N: nonce, A: AAD, P: plaintext, C: concatenation of ciphertext and tag
```

Since the test vector is not an official one, I checked it with BouncyCastle/Kotlin to be on the safe side: BouncyCastle/Kotlin returns the value for C for these K, N, A, P and thus confirms this test vector.

Minimal, reproducible example (using PyCryptodome 3.15.0):

```py
from Crypto.Cipher import AES

key = bytes.fromhex("000102030405060708090A0B0C0D0E0F")
plaintext = bytes.fromhex("0001020304050607")

# 15 bytes nonce
nonce=bytes.fromhex("EEDDCCBBAA99887766554433221103")
cipher = AES.new(key, AES.MODE_OCB, nonce=nonce)
(ct, tag) = cipher.encrypt_and_digest(plaintext)
print(ct.hex())
print(tag.hex())
print(ct.hex() + tag.hex()) # 579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5

print()

# 14 bytes nonce
nonce=bytes.fromhex("EEDDCCBBAA998877665544332211")
cipher = AES.new(key, AES.MODE_OCB, nonce=nonce)
(ct, tag) = cipher.encrypt_and_digest(plaintext)
print(ct.hex())
print(tag.hex())
print(ct.hex() + tag.hex()) # 579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5
```

Output:

```none
579fe6476a16cbcd
c22f0665e2b0b132de42e1917b89a2a5
579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5

579fe6476a16cbcd
c22f0665e2b0b132de42e1917b89a2a5
579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5
```

The expected result for C in the case of the 15 bytes nonce would have been: `084E869570194BD25032FE9E5328E45D507E74F3366E20D2`
However, for C the result of the 14 bytes nonce is returned: `579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5`

To be on the safe side, I verified with BouncyCastle/Kotlin that the 14 bytes nonce `EEDDCCBBAA998877665544332211` indeed returns for C the value `579fe6476a16cbcdc22f0665e2b0b132de42e1917b89a2a5`.

S. also on StackOverflow: [Bouncy Castle is throwing "mac check in OCB failed" when trying to decrypt an AES message][4]

[1]: https://www.rfc-editor.org/rfc/rfc7253.html#appendix-A
[2]: https://gitlab.com/dkg/ocb-test-vectors
[3]: https://gitlab.com/dkg/ocb-test-vectors/-/blob/main/test-vector-1-nonce120.txt
[4]: https://stackoverflow.com/questions/73557729/bouncy-castle-is-throwing-mac-check-in-ocb-failed-when-trying-to-decrypt-an-ae

**Comment by Legrandin** [comment]
Thanks for flagging. Fixed with 180d9d3a2740b18ce56bf4b81374f483b875607d

**Comment by Legrandin** [comment]
Fixed in v3.17
